### PR TITLE
Use CircleCI instead of Travis

### PR DESCRIPTION
**Pull request by JoeSouthan**
Let's standardise on CircleCI for all our projects

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      #     command: bundle exec rubocop --parallel --extra-details --display-style-guide
+
+workflows:
+  default:
+    jobs:
+      - test:
+          name: Ruby << matrix.ruby-version >>
+          matrix:
+            parameters:
+              ruby-version: ["2.5.8", "2.6.6", "2.7.1"]
```
Comment (ivgiuliani) [inline .circleci/config.yml:37]: We need to keep testing the older versions for now. We can cut a new release and drop support for the older ones but I wouldn't do it as part of this PR (I believe we only support releases that are not EOL, see https://github.com/gocardless/statesman/blob/master/docs/COMPATIBILITY.md for an example)

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          name: Install dependencies
+          command: bundle install --clean --no-cache --path vendor/bundle --jobs=4 --retry=3
+      - save_cache:
+          key: bundle-v1-<< parameters.ruby-version >>-{{ checksum "business.gemspec" }}
+          paths:
+            - vendor/bundle
+      - run:
+          name: Run tests
+          command: bundle exec rspec
+      # - run:
```
Comment (ivgiuliani) [inline .circleci/config.yml:26]: ✂️ if we're not using this

File: .circleci/config.yml (hunk 1 above)
Comment (JoeSouthan) [reply .circleci/config.yml:37]: So as far as I know 2.4 has been EOL'd but I can add it back in here 

https://www.ruby-lang.org/en/news/2020/04/05/support-of-ruby-2-4-has-ended/

File: .circleci/config.yml (hunk 2 above)
Comment (ivgiuliani) [reply .circleci/config.yml:26]: That's fine but we shouldn't be committing code that's commented out and is not relevant for this PR

**Pull request merged**